lib/endpoints/volumes.js

```diff
@@ -110,6 +110,11 @@ function listVolumes(req, res, next) {
 
     if (req.params.predicate) {
         listVolumesParams.predicate = req.params.predicate;
+    } else if (!req.params.state) {
+        // PUBAPI-1371: When we have no state and no predicate, we should
+        // filter out deleted and failed volumes.
+        listVolumesParams.predicate
+            = '{"and":[{"ne":["state","deleted"]},{"ne":["state","failed"]}]}';
```

Comment:
nit: using `JSON.stringify(predicateObject)` and defining `predicateObject` as a JS object and not a string might be easier to write/read/change, but that's really fine if you like the current form better.